packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -257,323 +231,412 @@ describe("Garbage Collection Tests", () => {
             );
         });
 
-        /**
-         * Here, the base snapshot contains nodes that are inactive and the GC blob in snapshot is in old format. The
-         * test validates that we generate inactive events for these nodes.
-         */
-        it("generates events for nodes that are inactive on load - old snapshot format", async () => {
-            // Create GC details for node 3's GC blob whose unreferenced time was > inactiveTimeoutMs ago.
-            // This means this node should become inactive as soon as its data is loaded.
-            const node3GCDetails: IGarbageCollectionDetailsBase = {
-                gcData: { gcNodes: { "/": [] } },
-                unrefTimestamp: Date.now() - (inactiveTimeoutMs + 100),
-            };
-            const node3Snapshot = getDummySnapshotTree();
-            const gcBlobId = "node3GCDetails";
-            const attributesBlobId = "attributesBlob";
-            node3Snapshot.blobs[gcBlobKey] = gcBlobId;
-            node3Snapshot.blobs[dataStoreAttributesBlobName] = attributesBlobId;
-
-            // Create a base snapshot that contains snapshot tree of node 3.
-            const baseSnapshot = getDummySnapshotTree();
-            baseSnapshot.trees[nodes[3].slice(1)] = node3Snapshot;
-
-            // Set up the getNodeGCDetails function to return the GC details for node 3 when asked by garbage collector.
-            const gcBlobMap = new Map([
-                [gcBlobId, node3GCDetails],
-                [attributesBlobId, {}],
-            ]);
-            const garbageCollector = createGarbageCollector({ baseSnapshot }, gcBlobMap);
-
-            // Remove node 3's reference from node 2 so that it is still unreferenced. The GC details from the base
-            // summary is not loaded until the first time GC is run, so do that immediately.
-            defaultGCData.gcNodes[nodes[2]] = [];
-            await garbageCollector.collectGarbage({ runGC: true });
+        describe("Sweep ready events", () => {
+            const snapshotCacheExpiryMs = 500;
+            const sweepTimeoutMs = defaultSessionExpiryDurationMs + snapshotCacheExpiryMs + oneDayMs;
 
-            // Validate that changed and loaded events are logged for node 3 since it is inactive.
-            garbageCollector.nodeUpdated(nodes[3], "Changed", Date.now(), testPkgPath);
-            garbageCollector.nodeUpdated(nodes[3], "Loaded", Date.now(), testPkgPath);
-            await garbageCollector.collectGarbage({});
-            assert(
-                mockLogger.matchEvents([
-                    { eventName: changedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg },
-                    { eventName: loadedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg },
-                ]),
-                "inactive events not logged as expected",
-            );
+            beforeEach(() => {
+                injectedSettings[runSessionExpiryKey] = "true";
```

Comment:
If we've got sessionExpiry on, shouldn't containers be closing as we advance the sweepTimeoutMs?  Why aren't they?

Reply:
These are unit tests and there is no container. GarbageCollector calls into the IGarbageCollectorRuntime->close and its a no op in these tests.